zephyr/lib/regions_mm.c

```diff
@@ -54,7 +53,7 @@ struct vmh_heap *vmh_init_heap(const struct vmh_heap_config *cfg,
 	 * available cores
 	 */
 	if (memory_region_attribute == MEM_REG_ATTR_CORE_HEAP) {
-		new_heap->virtual_region = &virtual_memory_regions[core_id];
+		new_heap->virtual_region = &virtual_memory_regions[0];
```

Comment:
Keeping the loop over all cores causes this function to be called once per each core, so multiple heaps will point to the same region? Is this correct?